src/witan/send/vis/ingest/transitions.clj

```diff
@@ -17,36 +16,39 @@
     (with-open [w (io/writer file-name)]
       (csv/write-csv w (into [header] (mapv extraction-fn transitions))))))
 
-(defn transition-calendar-years [transitions]
-  (into (sorted-set)
-        (map :calendar-year)
-        transitions))
+(defn transition-s1->census [{:keys [calendar-year setting-1 need-1 academic-year-1]}]
+  {:anon-ref -1
+   :calendar-year calendar-year
+   :setting setting-1
+   :need need-1
+   :academic-year academic-year-1})
 
-(defn max-transitions-calendar-year [transitions]
-  (apply max (transition-calendar-years transitions)))
+(defn transition-s2->census [{:keys [calendar-year setting-2 need-2 academic-year-2]}]
+  {:anon-ref -1
+   :calendar-year (inc calendar-year)
+   :setting setting-2
+   :need need-2
+   :academic-year academic-year-2})
 
-(defn max-census-year-from-transitions [transitions]
-  (inc (max-transitions-calendar-year transitions)))
+(defn transition->census-like [first-transition-year transition]
+  (if (= first-transition-year (:calendar-year transition))
+    [(transition-s1->census transition)
+     (transition-s2->census transition)]
+    [(transition-s2->census transition)]))
```

Comment:
why `transition-s2->census` twice?

Reply:
If it is the first year we need to output the s1 and s2 sides as census records. For all other years we just output the s2 side.